portal-kernel/src/com/liferay/portal/kernel/util/SessionClicks.java

```diff
@@ -145,6 +146,42 @@ public static void put(HttpSession httpSession, String key, String value) {
 	public static void put(
 		HttpSession httpSession, String namespace, String key, String value) {
 
+		if ((key.length() > _SESSION_CLICKS_MAX_SIZE_TERMS) ||
```

Comment:
@dantewang I do not like the same checking logic duplicated in the same class. Let's try to merge `put(HttpSession session, String key, String value) `and `put(HttpServletRequest httpServletRequest, String key, String value) `into one method. You should be able to pass in a function to deal with the difference between `HttpServletRequest` and `HttpSession`